src/dev.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useCallback } from 'react';
+import ReactDOM from 'react-dom';
+import Modal from './index';
+
+const App: React.FC = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const handleOpenModal = useCallback(() => {
+    setModalOpen(true);
+  }, []);
+
+  const handleCloseModal = useCallback(() => {
+    setModalOpen(false);
+  }, []);
```

Comment:
Tu peux simplifier comme ça je pense.
```suggestion
  const handleChangeModal = useCallback(() => {
    setModalOpen((prevState) => !prevState);
  }, [setModalOpen]);
```